optuna/distributions.py

```diff
@@ -301,6 +312,26 @@ def _contains(self, param_value_in_internal_repr):
         index = int(param_value_in_internal_repr)
         return 0 <= index and index < len(self.choices)
 
+    @staticmethod
+    def _to_valid_choice(choice):
+        # type: (Any) -> Union[None, bool, float, str]
+
+        if choice is None or isinstance(choice, (bool, float, str)):
+            return choice
+        try:
+            choice = float(choice)
+        except TypeError:
+            message = (
+                "Choices for a categorical distribution should be a tuple of None, bool, float, "
+                "str and castable to float for persistent storage but contains {} which is of "
+                "type {}.".format(choice, type(choice).__name__))
+            warnings.warn(message)
+
+            logger = logging._get_library_root_logger()
```

Comment:
In the Optuna convention, it's typical to use a module-specific logger instead of the library root logger.
```suggestion
            logger = logging.get_logger(__name__)
```

Reply:
Ah, I wasn't aware of that. Thank you. That applies to the logging in [`FrozenTrial.trial_id`](https://github.com/optuna/optuna/blob/master/optuna/structs.py#L231) and [`StudySummary.study_id`](https://github.com/optuna/optuna/blob/master/optuna/structs.py#L335) as well?

Reply:
> That applies to the logging in `FrozenTrial.trial_id` and `StudySummary.study_id` as well?

Yes, I think so.

Reply:
Created a small patch. https://github.com/optuna/optuna/pull/768

Reply:
👍